kDriveCore/Data/Models/DriveError.swift

```diff
@@ -155,6 +252,12 @@ public struct DriveError: Error, Equatable {
     static func serverError(statusCode: Int) -> DriveError {
         return errorWithUserInfo(.serverError, info: [.status: ErrorUserInfo(intValue: statusCode)])
     }
+    
+    /// Wraps a specific detailed error into a user facing localized DriveError
+    public mutating func wrapping(_ underlyingError: Error) -> Self {
+        self.underlyingError = underlyingError
```

Comment:
Maybe there is a built-in property embedded in the Error protocol for this use case ?

Reply:
Since the `DriveError` is implementing `Error` and not a subclass of `NSError`, we do not have it for free.
We could switch to a NSError, but that is less swifty. I can do both, let me know what you prefer.